File: src/lessons/welcome/bat/bool2/SortaSumEntity.java
package lessons.welcome.bat.bool2;

import static plm.core.ValueSerializer.deserialize;
import static plm.core.ValueSerializer.serialize;

import plm.universe.bat.BatEntity;

public class SortaSumEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(sortaSum((Integer)param[0], (Integer)param[1])));
    }
  }

  /* BEGIN TEMPLATE */
  int sortaSum(int a, int b)
  {
    /* BEGIN SOLUTION */
    int sum = a + b;
    if (sum >= 10 && sum <= 19)
      return 20;
    else
      return sum;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/bat/bool2/SortaSumEntity.scala
package lessons.welcome.bat.bool2

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest

class SortaSumEntity extends BatEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(sortaSum(param(0).asInstanceOf[Int], param(1).asInstanceOf[Int]) ))
	  }
	}

	/* BEGIN TEMPLATE */
	def sortaSum(a:Int, b:Int):Int = {
		/* BEGIN SOLUTION */
		val sum = a+b
		if (sum >= 10 && sum <= 19)
			return 20
		else
			return sum
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}
